smartconfig/_registry.py

```diff
@@ -0,0 +1,120 @@
+"""
+Internal register used to reference entries.
+
+Attributes:
+    global_configuration: The full configuration loaded from the YAML file along with all default values for entries.
+    used_paths: A set of paths already containing an entry.
+    overwritten_attributes: A set of paths that have been overwritten by a configuration file.
+    mapping_cache: A mapping of dot-delimited paths to their corresponding mappings in the config.
+"""
+
+import copy
+from typing import Any, Dict, Hashable, Mapping, MutableMapping, Set
+
+import smartconfig
+from .exceptions import ConfigurationError, ConfigurationKeyError
+
+global_configuration: Dict[Hashable, Any] = {}
+used_paths: Set["smartconfig.ConfigEntry"] = set()
+overwritten_attributes: Set[str] = set()
+mapping_cache: Dict[str, Mapping] = {}
+
+
+def _get_child_node(node_name: str, root: Any):
+    """
+    Retrieve the value of the node `node_name` which is a child of the `root` node.
+
+    Args:
+        node_name: The name of the child node to retrieve.
+        root: The parent node of the node to retrieve.
+
+    Returns:
+        The value of the child node.
+
+    Raises:
+        ConfigurationError: The parent is not a mapping node.
+        ConfigurationKeyError: The child node does not exist.
+    """
+    if not isinstance(root, Mapping):
+        raise ConfigurationError(f"Cannot retrieve node {node_name!r}: its parent is not a mapping node.")
+
+    if node_name not in root:
+        raise ConfigurationKeyError(f"Cannot retrieve node {node_name!r}: it does not exist.")
+
+    node = root[node_name]
+
+    # Use __get__ if the node is a descriptor.
+    if hasattr(node, '__get__'):
+        node = node.__get__()
+
+    return node
+
+
+def unload_defaults(path: str) -> None:
+    """
+    Remove the default values of all children of `path`.
+
+    Args:
+        path: The dot-delimited path to the parent node.
+    """
+    try:
+        node = get_node(path)
+    except (ConfigurationError, ConfigurationKeyError):
+        return
+
+    if not isinstance(node, MutableMapping):
+        return
```

Comment:
Should this be silently failing or raising an error. All entries _should_ be mapping nodes, right?

Reply:
Resolved by 4c58ba5